### PR TITLE
boards: nrf9160_pca10090: Clean up default configurations

### PR DESCRIPTION
**Pull request by anangl**
This PR contains two patches that remove a couple of entries from default configurations for the `nrf9160_pca10090` board. These entries do not make much sense for this board, as the corresponding features are not present in the nRF9160 chip.

**drivers: entropy: nrf5: Fix dependency of the enabling Kconfig option**

This driver makes use of the nRF RNG peripheral, so it can be used only
for SoCs that are equipped with one, and not all nRF SoCs are.
The option enabling the driver should then depend on `HAS_HW_NRF_RNG`,
which indicates the presence of this peripheral in a given SoC.

This patch removes also entries disabling this driver in default
configurations for nRF9160 SoC, as these were needed only because
of the invalid dependency of the ENTROPY_NRF5_RNG option.

A minor adjustment of Kconfig files of the nrf52_bsim board was
required as well, so that this board's configuration can properly
handle this corrected dependency.

**soc: nrf9160: Remove unsupported Kconfig option GPIO_AS_PINRESET**

In nRF9160 the reset pin is a dedicated one, it cannot be configured
as a regular GPIO pin, so this option should not be presented to users
building for this SoC, to not generate confusion.

### CODE REVIEW

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by ioannisg** [comment]
@aescolar could you take a quick look at this trivial patch?

**Review by aescolar — COMMENTED** [review]
@ioannisg I did. I'm not extremely happy about duplicating the definition of symbols into the nrf52_bsim. But I won't block this PR because of that.

-----------
This was the discussion I had with Andrzej on the matter:
```
Alberto Escolar
what do you think about moving all this nordic HAS_HW_NRF* and the SOC_COMPATIBLE* definitions to a soc/Kconfig.nordic?
that would be included from soc/Kconfig
the issue is that the soc/arm/nordic_nrf/Kconfig files are only included if ARM is selected otherwise they are not parsed at all
so for these symbols to be visible from the simulated boards their definitions need to be parsed always 
Andrzej Głąbek [2:33 PM]
I see
yes, for me it would be acceptable
Alberto Escolar [2:34 PM]
it would be cleaner I think.. and if tomorrow you build a SOC with another CPU architecture, you would be already ready to reuse them :slightly_smiling_face:
Andrzej Głąbek [2:37 PM]
I'm not sure what @ulfalizer and others would say on parsing all these symbols for all architectures when these are relevant only for Nordic SoCs and for the simulated board
I see we've got already something similar with SOC_COMPATIBLE_NRF options placed additionally in soc/Kconfig, but these are only three...
I guess the question is what is less ugly: the hack I pushed in my PR or such solution (edited)

**Comment by aescolar** [comment]
@anangl I remove the "Meta" tag as that was supposed to be used for "issues" which are used to collect other issues

**Comment by ulfalizer** [comment]
Only looked at it quickly, so tell me if I'm misunderstanding.

if a symbol needs to be a seen by a large part of the tree, even its for a kinda "hackish" reason, then I think it's best to put a single definition somewhere high up, like @aescolar suggested.

That's also a good spot to add a comment that explains why it needs to be there.

**Comment by ioannisg** [comment]
> 
> 
> Only looked at it quickly, so tell me if I'm misunderstanding.
> 
> If a symbol needs to be a seen by a large part of the tree, even its for a kinda "hackish" reason, then I think it's best to put a single definition somewhere high up, like @aescolar suggested.
> 
> That's also a good spot to add a comment that explains why it needs to be there.

In this particular case, I don't think we have many options. HAS_HW_NRF_RNG cannot be (naturally) defined elsewhere than in soc/arm/nordic_nrf

And if the nrf52_bsim board needs it, we need to make that visible - or re-define it
What do you think @ulfalizer ?

**Comment by anangl** [comment]
@aescolar @ulfalizer @ioannisg
What about modifying the PR [this way](https://github.com/anangl/zephyr/commit/30a5e38cba9b4bfec4a6ec4748a97985e47ee681), i.e. adding the read of `soc/arm/nordic_nrf/Kconfig.peripherals` to `boards/posix/nrf52_bsim/Kconfig`, so that `HAS_HW_NRF_RNG` could be simply selected in `boards/posix/nrf52_bsim/Kconfig.board`? I think it is less ugly than duplicating the definition of `HAS_HW_NRF_RNG`, but still all the `HAS_HW_NRF*` symbols don't need to be parsed for all architectures (what I see as a downside of what @aescolar suggested), only when the nrf52_bsim board is actually selected.

**Comment by ioannisg** [comment]
> 
> 
> @aescolar @ulfalizer @ioannisg
> What about modifying the PR [this way](https://github.com/anangl/zephyr/commit/30a5e38cba9b4bfec4a6ec4748a97985e47ee681), i.e. adding the read of `soc/arm/nordic_nrf/Kconfig.peripherals` to `boards/posix/nrf52_bsim/Kconfig`, so that `HAS_HW_NRF_RNG` could be simply selected in `boards/posix/nrf52_bsim/Kconfig.board`? I think it is less ugly than duplicating the definition of `HAS_HW_NRF_RNG`, and still all the `HAS_HW_NRF*` symbols don't need to be parsed for all architectures (what I see as a downside of what @aescolar suggested), only when the nrf52_bsim board is actually selected.

I am OK with that. @aescolar pls, comment :)

**Comment by aescolar** [comment]
@ioannisg @anangl I think that if the discussion drags for to long, you can merge this as it is if you are in a hurry to get it for 2.0 

> What about modifying the PR this way..

Ok, it is less ugly (I would add the source line at the top of the file instead of the bottom though).
But I would certainly want @ulfalizer 's blessing on this.

**Comment by ioannisg** [comment]
>Ok, it is less ugly (I would add the source line at the top of the file instead of the bottom though).
But I would certainly want @ulfalizer 's blessing on this.

Alright, let's see what @ulfalizer has to say about that.

**Comment by anangl** [comment]
I updated the PR as proposed above, with applied @aescolar's suggestion regarding moving the `source` line.

**Review by ioannisg — APPROVED** [review]

File: boards/posix/nrf52_bsim/Kconfig.board [hunk 1]
```diff
@@ -10,6 +10,10 @@ config BOARD_NRF52_BSIM
 	select CONSOLE_HAS_DRIVER
 	select NRF_RTC_TIMER
 	select CLOCK_CONTROL
+	# Pretend that the nRF RNG peripheral is present, so that the driver
```
Comment (aescolar) [inline boards/posix/nrf52_bsim/Kconfig.board]: Silly thing: You can remove the "Pretend that the nRF RNG peripheral is present , " part of this comment.
The RNG HW model is present :)

**Review by aescolar — APPROVED** [review]

File: boards/posix/nrf52_bsim/Kconfig.board (hunk 1 above)
Comment (anangl) [reply boards/posix/nrf52_bsim/Kconfig.board]: Right. What about changing it to "Indicate that the nRF RNG peripheral (actually a model of it ) is present, ..."?

Comment (aescolar) [reply boards/posix/nrf52_bsim/Kconfig.board]: Fine by me

Comment (anangl) [reply boards/posix/nrf52_bsim/Kconfig.board]: Changed.

**Review by ulfalizer — APPROVED** [review]
I'd still prefer to just move things up so that they're always visible wherever they need to be, even if it means e.g. putting stuff in the top-level Kconfig file. Parsing the entire tree (e.g. for doc generation) is a mess now, because of all the funky double definition stuff going on. It also messes up the doc output.

If a hack is needed, I think it's better to make it stand out loud and clear than to pretend it isn't there and try to sweep it under the rug. :)

Won't block over it though.

**Pull request merged**